### PR TITLE
feat: contenttype fields

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |n|
|New feature?   |y|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

New contentType fields property like contentType roles, plus removed unused columns.

After this PR and with the previous roles refactor:
- the entity ContentType went from **1908** lines to **1195**
- the table contenttype went from **60** columns to **33**

https://github.com/ems-project/ems-project.github.io/pull/2

### CODE REVIEW

File: src/Resources/views/contenttype/edit.html.twig [hunk 1]
```diff
@@ -66,7 +66,6 @@
                             {% if contentType.environment.managed %}
                                 {{ form_row(form.orderField) }}
                             {% endif %}
-                            {# 						{{ form_row(form.categoryField) }}#}
```
Comment (Davidmattei) [inline src/Resources/views/contenttype/edit.html.twig:69]: check line 63 (already printed)

File: src/Resources/DoctrineMigrations/Scripts/ScriptContentTypeFields.php [hunk 2]
```diff
@@ -26,18 +26,18 @@ public function scriptEncodeFields(AbstractMigration $migration): void
             ]);
         }
 
-        $migration->addSql('ALTER TABLE content_type DROP labelField');
-        $migration->addSql('ALTER TABLE content_type DROP circles_field');
-        $migration->addSql('ALTER TABLE content_type DROP business_id_field');
-        $migration->addSql('ALTER TABLE content_type DROP color_field');
```
Comment (Davidmattei) [inline src/Resources/DoctrineMigrations/Scripts/ScriptContentTypeFields.php]: fixed in next commit

File: src/Resources/DoctrineMigrations/Scripts/ScriptContentTypeFields.php [hunk 3]
```diff
@@ -26,18 +26,18 @@ public function scriptEncodeFields(AbstractMigration $migration): void
             ]);
         }
 
-        $migration->addSql('ALTER TABLE content_type DROP labelField');
```
Comment (Davidmattei) [inline src/Resources/DoctrineMigrations/Scripts/ScriptContentTypeFields.php]: Script fixed in next commit

File: src/Entity/ContentType.php [hunk 4]
```diff
@@ -207,18 +207,6 @@ class ContentType extends JsonDeserializer implements \JsonSerializable, EntityI
      */
     public $defaultValue;
 
-    /**
-     * @var string
-     *
-     * @ORM\Column(name="translationField", type="string", length=100, nullable=true)
-     */
-    protected $translationField;
```
Comment (Davidmattei) [inline src/Entity/ContentType.php:333]: Added in https://github.com/ems-project/EMSCoreBundle/pull/282

File: src/Core/ContentType/ContentTypeFields.php [hunk 5]
```diff
@@ -36,7 +42,7 @@ public function __construct(array $data)
     public function getFields(): array
     {
         /** @var array<string, ?string> $cleaned */
-        $cleaned = ArrayHelper::map($this->fields, fn ($v) => (\strlen($v) > 0 ? $v : null));
+        $cleaned = ArrayHelper::map($this->fields, fn (?string $v) => (null !== $v && \strlen($v) > 0 ? $v : null));
```
Comment (theus77) [inline src/Core/ContentType/ContentTypeFields.php:53]:
```suggestion
        $cleaned = ArrayHelper::map($this->fields, fn (?string $v) => (empty($v) ? null : $v));
```

File: src/Resources/DoctrineMigrations/Scripts/ScriptContentTypeFields.php [hunk 6]
```diff
@@ -9,26 +9,42 @@ trait ScriptContentTypeFields
 {
     public function scriptEncodeFields(AbstractMigration $migration): void
     {
+        $emptyStringToNull = function (?string $value): ?string {
+            return $value && \strlen($value) > 0 ? $value : null;
```
Comment (theus77) [inline src/Resources/DoctrineMigrations/Scripts/ScriptContentTypeFields.php:13]:
```suggestion
            return empty($value) ? null : $value;
```

**Review by theus77 — APPROVED** [review]

**Pull request merged**